src/main/scala/com/codacy/duplication/pmd/Cpd.scala

```diff
@@ -0,0 +1,146 @@
+package com.codacy.duplication.pmd
+
+import java.io.{ByteArrayOutputStream, PrintStream}
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Path, Paths}
+import java.util
+
+import better.files.File
+import codacy.docker.api.duplication._
+import codacy.docker.api.{DuplicationConfiguration, Source}
+import com.codacy.api.dtos.{Language, Languages}
+import net.sourceforge._
+import net.sourceforge.pmd.cpd
+import net.sourceforge.pmd.cpd.Match
+import play.api.libs.json.{JsNumber, JsValue}
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+object Cpd extends DuplicationTool {
+
+  private val allLanguages: List[Language] =
+    List[Language](
+      Languages.Python,
+      Languages.Ruby,
+      Languages.Java,
+      Languages.Javascript,
+      Languages.Scala,
+      Languages.CSharp)
+
+  override def apply(
+    path: Source.Directory,
+    language: Option[Language],
+    options: Map[DuplicationConfiguration.Key, DuplicationConfiguration.Value]): Try[List[DuplicationClone]] = {
+
+    val baos = new ByteArrayOutputStream()
+    System.setErr(new PrintStream(baos, true, "utf-8"))
+
+    val directoryPath: Path = (File.currentWorkingDirectory / path.path).path
+
+    getLanguages(language).map { languages =>
+      runCPD(directoryPath, languages, options)
+    } match {
+      case Success(matches) => Success(matches)
+      case Failure(e) =>
+        val errString = new String(baos.toByteArray, StandardCharsets.UTF_8)
+        val msg =
+          s"""|Failed to execute duplication: ${e.getMessage}
+              |std:
+              |$errString
+         """.stripMargin
+        baos.close()
+
+        Failure(new Exception(msg, e))
+    }
+  }
+
+  private def runCPD(
+    directory: Path,
+    languages: List[(pmd.cpd.Language, Int)],
+    options: Map[DuplicationConfiguration.Key, DuplicationConfiguration.Value]): List[DuplicationClone] = {
+
+    languages.flatMap {
+      case (lang, minTokenMatch) =>
+        val configuration = getConfiguration(lang, minTokenMatch, options)
+        runWithConfiguration(configuration, directory)
+    }
+
+  }
+
+  private def runWithConfiguration(config: pmd.cpd.CPDConfiguration, directory: Path): List[DuplicationClone] = {
+    val cpd = new pmd.cpd.CPD(config)
+    cpd.addRecursively(directory.toFile)
+
+    cpd.go()
+
+    val res: util.Iterator[Match] = cpd.getMatches
+    res.asScala.map(duplicationClone(_, directory)).toList
+  }
+
+  private def getLanguages(language: Option[Language]): Try[List[(pmd.cpd.Language, Int)]] = {
+
+    val languagesOpt: Option[Try[List[(cpd.Language, Int)]]] =
```

Comment:
Lets make methods smaller and simpler.
Why does getLanguages return languages configuration? Separate the two to make the methods smaller.
Why are we returning a tuple? Do a case class to avoid wierd behaviour.